File: benchmark/corpus/csharp/cases/vuln_cors.cs
public class VulnCors
{
    public void Handle()
    {
        var origin = Request.QueryString["origin"];
        // ZS-CS-018: CORS misconfiguration — tainted origin reflected into Access-Control-Allow-Origin
        Response.AppendHeader("Access-Control-Allow-Origin", origin);
    }
}

File: benchmark/corpus/java/cases/vuln_cors.java
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

public class VulnCors {
    void handle(HttpServletRequest request, HttpServletResponse response) throws Exception {
        String origin = request.getParameter("origin");
        // ZS-JAVA-023: CORS misconfiguration — tainted origin reflected into Access-Control-Allow-Origin
        response.setHeader("Access-Control-Allow-Origin", origin);
    }
}
